src/pods/test-sentences/test-sentences.container.tsx

```diff
@@ -50,14 +47,41 @@ export const TestSentencesContainer: React.FC = () => {
     );
   }, []);
 
+  const pickRandomVerb = (selectedVerbs: string[]): VerbEntityApi => {
+    if (Array.isArray(selectedVerbs)) {
+      const selectedVerbsLength = selectedVerbs.length;
+      const index = Math.floor(Math.random() * selectedVerbsLength);
+      const [selectedVerbWithInfo] = verbCollection.filter(
+        verb => verb.infinitive === selectedVerbs[index]
+      );
+      return selectedVerbWithInfo;
+    }
+    return {
+      infinitive: '',
+      past: '',
+      participle: '',
+      translation: '',
+    };
+  };
+
+  const randomVerb = pickRandomVerb(selectedVerbs);
+
   const mapRandomSentence = (
     sentencesCollection: SentenceEntityApi[]
-  ): SentenceEntityVm =>
-    mapFromSentenceApiToSentenceVm(
-      pickRandomSentence(sentencesCollection),
-      selectedVerbsWithInfo
+  ): SentenceEntityVm => {
+    const sentencesWithVerbSelected = sentencesCollection.filter(
+      sentence => sentence.verb === randomVerb.infinitive
+    );
+    const randomSentenceWithVerbSelected = pickRandomSentence(
+      sentencesWithVerbSelected
     );
 
+    return mapFromSentenceApiToSentenceVm(
+      randomSentenceWithVerbSelected,
+      randomVerb
+    );
+  };
```

Comment:
Same here separate this into test-senteces.business.ts

Reply:
done.